cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -73,9 +75,10 @@
   public HBaseTable(String name, ConflictDetection level, Configuration hConf, boolean enableReadlessIncrements)
     throws IOException {
     super(name, level, enableReadlessIncrements);
-
     hTableName = HBaseTableUtil.getHBaseTableName(name);
-    HTable hTable = new HTable(hConf, hTableName);
+    HBaseTableUtil tableUtil = new HBaseTableUtilFactory().get();
+    TableId tableId = TableId.from(hTableName);
```

Comment:
ok this comes straight from the dataset spec.  It seems wrong that we're encoding the namespace inside the name in the spec. shouldn't it be it's own field in the spec or passed in by whoever is using the spec?